CMakeLists.txt

```diff
@@ -142,6 +148,20 @@ else()
   endif()
 endif()
 
+if(UNIX AND NOT APPLE)
```

Comment:
Could you add the `CMAKE_CXX_COMPILER MATCHES "clang"` to this conditional too? It seems more consistent with the rest of the CMake logic changes.